plugins/examples/rescore/src/main/java/org/elasticsearch/example/rescore/ExampleRescoreBuilder.java

```diff
@@ -163,34 +163,36 @@ public TopDocs rescore(TopDocs topDocs, IndexSearcher searcher, RescoreContext r
                  * them in (reader, field, docId) order because that is the
                  * order they are on disk.
                  */
-                ScoreDoc[] sortedByDocId = new ScoreDoc[topDocs.scoreDocs.length];
-                System.arraycopy(topDocs.scoreDocs, 0, sortedByDocId, 0, topDocs.scoreDocs.length);
+                ScoreDoc[] sortedByDocId = new ScoreDoc[end];
+                System.arraycopy(topDocs.scoreDocs, 0, sortedByDocId, 0, end);
                 Arrays.sort(sortedByDocId, (a, b) -> a.doc - b.doc); // Safe because doc ids >= 0
                 Iterator<LeafReaderContext> leaves = searcher.getIndexReader().leaves().iterator();
                 LeafReaderContext leaf = null;
                 SortedNumericDoubleValues data = null;
                 int endDoc = 0;
                 for (int i = 0; i < end; i++) {
-                    if (topDocs.scoreDocs[i].doc >= endDoc) {
+                    ScoreDoc scoreDoc = sortedByDocId[i];
+                    if (scoreDoc.doc >= endDoc) {
                         do {
                             leaf = leaves.next();
                             endDoc = leaf.docBase + leaf.reader().maxDoc();
-                        } while (topDocs.scoreDocs[i].doc >= endDoc);
+                        } while (scoreDoc.doc >= endDoc);
                         LeafFieldData fd = context.factorField.load(leaf);
-                        if (false == (fd instanceof LeafNumericFieldData)) {
+                        if (!(fd instanceof LeafNumericFieldData)) {
```

Comment:
`== false` is preferred. Though I think the example plugins don't fall under our formatting rules.

Reply:
Reverted back to `== false` format.